updated FAQs (#14249)
docker/docker.github.io: single-sign-on/faqs.md

toc_max: 2
---

## General SSO overview

### Q: Is Docker SSO available for all paid subscriptions?

Docker Single Sign-on (SSO) is only available with the Docker Business subscription. Upgrade your existing subscription to start using Docker SSO.

### Q: How does Docker SSO work?

Docker Single Sign-on (SSO) allows users to authenticate using their identity providers (IdPs) to access Docker. Docker currently supports Azure AD and any SAML 2.0 identity providers. When you enable SSO, users are redirected to your provider’s authentication page to authenticate using their email and password.

### Q: What SSO flows are supported by Docker?

Docker currently supports Service Provider Initiated (SP-initiated) SSO flow. This means users must sign in to Docker Hub or Docker Desktop to initiate the SSO authentication process.

### Q: Where can I find detailed instructions on how to configure Docker SSO?

> You first need to establish a SSO connection with your identity provider, and the company email domain needs to be verified prior to SSO enforcement for your users. For detailed step-by-step instructions on how to configure Docker SSO, see [Single Sign-on](index.md).

Reviewer: an SSO connection

Author: updated

### Q: Does Docker SSO support multi-factor authentication (MFA)?

When an organization uses SSO, MFA is determined at the IdP level, not by the Docker system.

### Q: Do I need a specific version of Docker Desktop for SSO?

> Yes, all users in your organization must upgrade to Docker Desktop version 4.4.0 or higher. Users on older versions of Docker Desktop will not be able to sign in after enforcing SSO if the company domain email is used to log in or as the primary email associated with an existing Docker account Your users with existing accounts can not sign in with their username and password.

Reviewer: ```suggestion
Yes, all users in your organization must upgrade to Docker Desktop version 4.4.0 or higher. Users on older versions of Docker Desktop will not be able to sign in after enforcing SSO if the company domain email is used to log in or as the primary email associated with an existing Docker account. Your users with existing accounts cannot sign in with their username and password.
```

Author: Updated

You must provide an email address as an attribute to authenticate via SAML. The ‘Name’ attribute is currently optional.

### Q: Does the application recognize the NameID/Unique Identifier in the SAMLResponse Subject?

The preferred format is your email address, which should also be your Name ID.

### Q: When SAML SSO is enforced, at what stage is the login required to be tracked through SAML? At runtime or install time?

At runtime for Docker Desktop if it’s configured to require authentication to the organization.

### Q: How long is the grace-period for using regular user id and password for the docker desktop itself regardless of the enforced SSO?

Currently, we do not have a date on when the grace-period will end.

### Q: Do you have any information on how to use the Docker Desktop application in accordance with the SSO users we provide? How can we verify that we are handling the licensing correctly?

Verify that your users have downloaded the latest version of Docker Desktop. We will be enhancing user management observability and capabilities in the near future.

## Docker org and Docker ID

### Q: What’s a Docker ID? Can I retain my Docker ID when using SSO?

> For a personal Docker ID, a user is the account owner, it’s associated with access to the user's repositories, images, assets. An end user can choose to have a company domain email on the Docker account, when SSO is enforced, the account will be tied to the company’s organization account. Alternatively, when SSO is enforced for a company organization, any user logging in without an existing account using verified company domain email will automatically have an account provisioned, and a new Docker ID created.

Reviewer: > when SSO is enforced for a company organization,

Should we use either company or organization here?

Author: Updated with organization

### Q: If I have multiple orgs how will that affect my org if they are all connected to the same  domain?

We are currently limited in supporting such a setup, and would recommend setting up different teams under the same org if you plan to enforce SSO and only have one email domain.

## Identity providers

### Q: Is it possible to use more than one IdP with Docker SSO?

No. You can only configure Docker SSO to work with a single IdP. A domain can only be associated with a single IdP. Docker currently supports Azure AD and identity providers that support SAML 2.0

### Q Is it possible to change my identity provider after configuring SSO?

Yes. You must delete your existing IdP configuration in Docker Hub and follow the instructions to Configure SSO using your IdP. If you had already turned on enforcement, you should turn off enforcement before updating the provider SSO connection.

### Q: What information do I need from my Identity providers to configure SSO?

To enable SSO in Docker, you need the following from your IdP:

> * **SAML**: Entity ID, ACS URL, Single Logout URL and Certificate Download URL

Author: @technicallyjosh Updating here and also adding this to our SSO Docs

They can do it one time to add it to a connection. If they ever change idPs and have to set up SSO again, they will need to verify again.

### Q: Is adding Domain required to configure SSO? What domains should I be adding? And how do I add it?

Adding and verifying Domain is required to enable and enforce SSO. Click Add Domain and specify the email domains that are allowed to authenticate via your server. This should include all email domains users will use to access Docker. Public domains are not permitted, such as gmail.com, outlook.com, etc. Also, the email domain should be set as the primary email.

### Q: How can we send a verification e-mail to just the domain reference? Wouldn’t we need a full e-mail address?

### Q: What would the full content of the registry.json need to be? Can an example be provided?

### Q: Does the registry.json go on the user’s workstation?

### Q: If users are using their personal email, do they have to convert to using the Org’s domain before they can be invited to join an Org? Is this just a quick change in their Hub account?

> No, they do not. Though they can add multiple emails to a dockerid if they choose to. However, that email can only be used once across docker. The other thing to note is that (as of January 2022) SSO will not work for multi domains as a MVP and it will not work for personal emails either.

Reviewer: ```suggestion
No, they do not. Though they can add multiple emails to a Docker ID if they choose to. However, that email can only be used once across Docker. The other thing to note is that (as of January 2022) SSO will not work for multi domains as an MVP and it will not work for personal emails either.
```

### Q: Since Docker ID is tracked from SAML, at what point is the login required to be tracked from SAML? Runtime or install time?

Runtime for Docker Desktop if they configure Docker Desktop to require authentication to their org.

### Q: Do you support IdP-initiated authentication (e.g., Okta tile support)?

We do not support IdP-initiated authentication. Users must initiate login through Docker Desktop or Hub.

## SSO enforcement

### Q: Can I enable SSO in all organizations?

You can enable SSO on organizations that are part of the Docker Business subscription.

### Q: We currently have a Docker Team subscription. How do we enable SSO?

> Docker SSO is available with a Docker Business subscription. To enable SSO, you must first upgrade your subscription to a Docker Business subscription. To learn how to upgrade your existing account, see Upgrade your subscription.

Reviewer: Missing link to Upgrade your subscription

### Q: How does service accounts work with SSO?

Service accounts work like any other user when SSO is turned on. If the service account is using an email for a domain with SSO turned on, it needs a PAT for CLI and API usage.

### Q: Is DNS verification required to enable SSO?

Yes. You must verify a domain before using it with an SSO connection.

### Q: Does Docker SSO support authenticating through the command line?

> Yes. When SSO is enabled, you can access the Docker CLI through Personal Access Tokens (PATs).  Each user must create a PAT to access the CLI. To learn how to create a PAT, see Manage access tokens. Before we transition to PATs, CLI users can continue logging in using their personal credentials until early next year to mitigate the risk of interrupting CI/CD pipelines.

Reviewer: Missing link to the Manage Access tokens section

### Q: How does SSO affect our automation systems and CI/CD pipelines?

Before enforcing SSO, you must create PATs for automation systems and CI/CD pipelines and use the tokens instead of a password. When SSO is enforced, password-based authentication no longer works on your automation systems and CI/CD pipelines.

### Q: I have a user working on projects within Docker Desktop but authenticated with personal or no email. After they purchase DB licenses, they will implement and enforce SSO via Okta to manage their users. When this user signs on SSO, is their work on DD compromised/impacted with the migration to the new account?

> If they already have their splunk email on their account, then it will be migrated to SSO.

Reviewer: Can we update this answer to be generic? Not sure if we should mention Splunk here?

Author: updated

> ### Q: If an organization enables SSO, the owners can control DockerIDs associated with their work email domain. Some of these DockerIDs will not be users of Docker Desktop and therefore don't require a Business subscription. Can the owners choose which DockerIDs they add to their DockerOrg and get access to business features? Is there a way to flag which of these DockerIDs are Docker Desktop users?

Reviewer: ```suggestion
### Q: If an organization enables SSO, the owners can control Docker IDs associated with their work email domain. Some of these Docker IDs will not be users of Docker Desktop and therefore don't require a Business subscription. Can the owners choose which Docker IDs they add to their Docker org and get access to Business features? Is there a way to flag which of these Docker IDs are Docker Desktop users?
```

SSO enforcement will apply to any domain email user, and automatically add that user to the Docker Hub org that enables enforcement. The admin could remove users from the org manually, but those users wouldn't be able to authenticate if SSO is enforced.

### Q: Can I enable SSO and hold off on the domain verification and enforcement options?

> Yes, they can choose to not enforce, and people have the option to use either Docker ID (standard email/password) or email address (SSO) at the sign-in screen.Though hold off on the domain verification -- to be clear (maybe it's just the wrong language used in the question?), they'd still need to verify the domain (showing that they "own" the domain)

Reviewer: Can we remove the internal comment about domain verification and wrong language?

Author: Thank you! Updated

### Q: We have enforced SSO, but one of our users is connected to several organizations (and several email-addresses) and is able to bypass SSO and login via userid and password. Why is this happening?

> They can only have 1 email and this is because that email domain doesn’t match the enforced SSO. (if it’s not enforced, they can use un/pass)

Author: Updated

### Q: Is there a way to test this functionality in a test tenant with Okta before going to production?

> A test organization can be created. Companies can set up a new 5 seat Business Plan on a new organization to test with (making sure to only enable SSO, not enforce it or all domain email users will be forced to sign in to that test tenant).

Reviewer: ```suggestion
Yes, you can create a test organization. Companies can set up a new 5 seat Business plan on a new organization to test with (making sure to only enable SSO, not enforce it or all domain email users will be forced to sign in to that test tenant).
```

### Q: Once we enable SSO for Docker Desktop, what is the impact to the flow for Build systems that use service accounts?

If SSO is enabled, there is no impact for now. We'll continue to support either username/password or personal access token sign-in.

> However, if you **enforce** SSO:

Reviewer: Add a blank line before the list item

### Q: Is it possible to force users of Docker Desktop to authenticate, and/or authenticate using their company’s domain?

> Yes. Admins can force users to authenticate with Docker Desktop by provisioning a registry.json configuration file. The registry.json file will force users to authenticate as a user that is configured in the allowedOrgs list in the registry.json file.

Reviewer: Can we add a link to the registryjson config topic https://docs.docker.com/docker-hub/configure-sign-in/

> Yes. Admins can force users to authenticate with Docker Desktop by provisioning a registry.json configuration file. The registry.json file will force users to authenticate as a user that is configured in the allowedOrgs list in the registry.json file.

Author: Updated

### Q: Can someone join the organization without an invitation? Is it possible to put specific users to an organization with existing email accounts?

Not without SSO. Joining requires an invite from a member of the Owners group. When SSO is enforced, then the domains verified through SSO will allow users to automatically join the organization the next time they sign in as a user that has a domain email assigned.

### Q: When we send an invitation to the user, will the existing account be consolidated and retained?

Yes, the existing user account will join the organization with all assets retained.

### Q: How can I view, update, and remove multiple email addresses for my users?

We only support one email per user on the Docker platform.

### Q: How can I remove invitees to the org who have not signed in?

> They can go to the invitee list in the org view and remove them.

Reviewer: ```suggestion
Org owners can go to the **Invitees** tab on the organization page in Docker Hub and remove the invitees from the list. The Invitees tab appears on the organization page only when you have invited users to your org who do not have a Docker account or an ID yet.
```

Author: Updated

* Each user has created a PAT to replace their passwords to allow them to log in through Docker CLI
* Confirm that all CI/CD pipelines automation systems have replaced their passwords with PATs.

For detailed prerequisites and for instruction on how to enable SSO, see [Configure Single Sign-on](index.md).

### Q: What impact can users expect once we start onboarding them to SSO accounts?

When SSO is enabled and enforced, your users just have to sign in using the email address and password.

### Q: Is Docker SSO fully synced with Active Directory (AD)?

Docker doesn’t currently support a full sync with AD. That is, if a user leaves the organization, administrators must sign in to Docker Hub and manually [remove the user](/orgs#remove-team-mebers) from the organization.

Additionally, you can use our APIs to complete this process.

### Q: What is the best way to provision the Docker Subscription without SSO?

> Admins in the Owners group in the orgs can invite users via Hub UI, by email address (for any user) or by docker id (assuming the user has created a user account on Hub already).

Reviewer: ```suggestion
Admins in the Owners group in the orgs can invite users through Docker Hub UI, by email address (for any user) or by Docker ID (assuming the user has created a user account on Hub already).
```

### Q: If we do this manually the first time, Is it possible to register for example max.mustermann@zeiss.com in the Dashboard and he will get an invitation link per email?

Yes, if the user is added via email address to an org, they will receive an email invite.If invited via docker ID as an existing user instead, they'll be added to the organization automatically. We'll be adding a new invite flow in the near future that will require an email invite in this situation as well (so the user can choose to opt out). If the org later sets up SSO for [zeiss.com](www.zeiss.com) domain, the user will automatically be added to the domain SSO org next sign in which requires SSO auth with the identity provider (Hub login will automatically redirect to the identity provider).

> ### Q: Can someone Join the Organization without an invitation? Is it possible to put specific users to our organization with existing email@zeiss.com accounts?

Reviewer: ```suggestion
### Q: Can someone join the Organization without an invitation? Is it possible to put specific users to an organization with existing email accounts?
```

> Not without SSO. Joining requires an invite from an Owners group member in the org.With SSO, then yes, whatever domains are claimed and verified will allow users to automatically join the organization the next time they sign in as a user that has a domain email assigned.

Reviewer: ```suggestion
Not without SSO. Joining requires an invite from a member of the Owners group. When SSO is enforced, then the domains verified through SSO will allow users to automatically join the organization the next time they sign in as a user that has a domain email assigned.
```

> ### Q: When we send an invitation to the user, will the existing max.mustermann@zeiss.com account be consolidated and retained?

Author: Thanks Usha, updated!

> Yes, the existing user account will join the organization. The user will be affected in certain ways (e.g. image access management policy settings, or the fact that SSO users can't change their email address), but otherwise with all assets retained.

Author: Rephrase this

### Q: How can I view, update, and remove multiple email addresses for my users?

> We only support one email in our system. The management of user info will come with SCIM.

Reviewer 2: ```suggestion
We only support one email per user on the Docker platform.
```

### Q: How can I remove invitees to the org who have not signed in?

> They can go to the invitee list in the org view and remove them

Reviewer 2: ```suggestion
They can go to the invitee list in the org view and remove them.
```
